library/camel-kamelets-catalog/src/main/resources/kamelets/message-timestamp-router-action.kamelet.yaml

```diff
@@ -0,0 +1,74 @@
+    camel.apache.org/provider: "Apache Software Foundation"
+    camel.apache.org/kamelet.group: "Actions"
+spec:
+  definition:
+    title: "Message Timestamp Router Action"
+    description: "Update the topic field as a function of the original topic name and the record's timestamp field."
+    required:
+      - timestampKeys
+    properties:
+      topicFormat:
+        title: Topic Format
+        description: Format string which can contain '$[topic]' and '$[timestamp]' as placeholders for the topic and timestamp, respectively.
+        type: string
+        default: "topic-$[timestamp]"
+      timestampFormat:
+        title: Timestamp Format
+        description: Format string for the timestamp that is compatible with java.text.SimpleDateFormat.
+        type: string
+        default: "yyyyMMdd"
+      timestampKeys:
+        title: Timestamp Keys
+        description: Comma separated list of Timestamp keys. The timestamp is taken from the first found field.
+        type: string
+      timestampKeyFormat:
+        title: Timestamp Keys Format
+        description: Format of the timestamp keys. Possible values are 'timestamp' or any format string for the timestamp that is compatible with java.text.SimpleDateFormat. In case of 'timestamp' the field will be evaluated as milliseconds since 1970, so as a UNIX Timestamp. 
```

Comment:
Until 3.10, where we can specify the parameter as optional. After that we can remove the timestamp workaround.